internal/pkg/aws/identity/identity.go

```diff
@@ -19,8 +20,11 @@ type Service struct {
 
 // New returns a Service configured with the input session.
 func New(s *session.Session) Service {
+	stsClient := sts.New(s)
+	stsClient.Handlers.Build.PushBackNamed(handlers.ArcherUserAgentHandler())
```

Comment:
Can you look into modifying the `aws/session` pkg instead of adding a handler to all other packages?
https://docs.aws.amazon.com/sdk-for-go/api/aws/session/#hdr-Adding_Handlers

This way we won't have to remember to add this handler anytime we define a new client. It will only be consumed in a single package.